### PR TITLE
Add support for nomad.env file / Add nomad 1.0.5 support

### PR DESCRIPTION
**Pull request by bastelfreak**
Nomad 1.0.5 ships a new systemd unit file with the following line:
```
EnvironmentFile=/etc/nomad.d/nomad.env
```
because there is no `-` before the leading `/`, the file is required.
Currently this module purges all unknown files in the nomad config
directory. This patch adds an `$env_vars` variable where people can pass
environment variables to nomad. They will be written to the nomad.env
file. The upstream packages provide the file:

```
reroot@server02 ~ # repoquery -l nomad | grep env
/etc/nomad.d/nomad.env
root@server02 ~ # rpm -qa | grep nomad
nomad-1.0.5-1.x86_64
root@server02 ~ #
```

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -148,6 +150,7 @@
   Boolean $manage_service                        = true,
   Boolean $manage_service_file                   = false,
   Boolean $restart_on_change                     = true,
+  Array[String[3]] $env_vars                     = [],
```
Comment (ekohl) [inline manifests/init.pp:153]: Why 3?

Comment (bastelfreak) [reply manifests/init.pp:153]: shortest possible env var would be something like `A=B`. I think a lower string length isn't possible.

Comment (ekohl) [reply manifests/init.pp:153]: Empty value to unset something? Also wondering if you want to make it a hash instead.

Comment (genebean) [reply manifests/init.pp:153]:
```puppet
  Optional[Array[String[3]]] $env_vars = [],
```

Might be more clear here.

Comment (ekohl) [reply manifests/init.pp:153]: That doesn't make sense. It's equivalent to:
```puppet
Variant[Undef, Array[String[3]]] $env_vars = [],
```

However, if you "call" it as such:
```puppet
class { 'nomad':
  env_vars => undef,
}
```

Then Puppet resolves it to what the default is, which is `[]`. Thus there is no way to pass in `undef` and `Optional` is just redundant.

Comment (bastelfreak) [reply manifests/init.pp]: I updated it to work with a hash. can you please take a look again?

**Comment by attachmentgenie** [comment]
```bash
system ❯ nomad -version          
Nomad v1.0.5 (0b870631cfa0c8e52cf698d7a7cc7989fbaec576)
system ❯ dpkg -L nomad | grep env
/etc/nomad.d/nomad.env
```

The [file](https://github.com/hashicorp/nomad/blob/main/dist/systemd/nomad.servicel) in the repo and described solution in the [docs](https://learn.hashicorp.com/tutorials/nomad/production-deployment-guide-vm-with-consul#configure-systemd) dont for now, nor was this mentioned in the release/changelog; very interesting, but i am seeing the same thing on ubuntu, so it must be real ;-).

**Review by attachmentgenie — APPROVED** [review]

File: manifests/config.pp [hunk 2]
```diff
@@ -28,4 +28,13 @@
...
     content => $_config,
   }
+  $content = join(map($nomad::env_vars) |$key, $value| { "${key}=${value}" }, "\n")
+  file { "${nomad::config_dir}/nomad.env":
+    ensure  => 'file',
+    owner   => 'root',
+    group   => 'root',
+    mode    => $nomad::config_mode,
+    content => "${content}\n",
+    require => File[$nomad::config_dir],
```
Comment (ekohl) [inline manifests/config.pp:38]: Isn't this done by default for file since Puppet 2.7?

**Pull request merged**